packages/doenetml-worker-rust/lib-js-wasm-binding/src/eval-math.ts

```diff
@@ -18,9 +24,128 @@ export function evalWithMathExpressionsInScope(source: string) {
                 "Running with MathExpressions in scope",
                 MathExpressions,
                 window,
+                serializedComponentsReviver,
             );
         }
-        return JSON.stringify(eval(source));
+
+        return JSON.stringify(eval(source), serializedComponentsReplacer);
     }
     return _eval(me, globalThis);
 }
+
+/**
+ * Parse the string `text` into math using the `math-expressions` text parser.
+ * Return the `JSON.stringify`ed results.
+ */
+export function parseTextIntoMath(
+    text: string,
+    splitSymbols: boolean,
+    functionSymbols: string[],
+    parseScientificNotation: boolean,
+) {
+    let fromText = textToMathFactory({
+        splitSymbols: splitSymbols,
+        parseScientificNotation: parseScientificNotation,
+        functionSymbols,
+    });
+    let expression;
+    try {
+        expression = fromText(text);
+    } catch (e) {
+        expression = me.fromAst("\uFF3F"); // long underscore
+        console.warn("Invalid value for a math of text format: " + text);
+    }
+
+    return JSON.stringify(expression, serializedComponentsReplacer);
+}
+
+/**
+ * Parse the string `latex` into math using the `math-expressions` latex parser.
+ * Return the `JSON.stringify`ed results.
+ */
+export function parseLatexIntoMath(
+    latex: string,
+    splitSymbols: boolean,
+    functionSymbols: string[],
+    parseScientificNotation: boolean,
+) {
+    let fromLatex = latexToMathFactory({
+        splitSymbols: splitSymbols,
+        parseScientificNotation: parseScientificNotation,
+        functionSymbols,
+    });
+    let expression;
+    try {
+        expression = fromLatex(latex);
+    } catch (e) {
+        expression = me.fromAst("\uFF3F"); // long underscore
+        console.warn("Invalid value for a math of latex format: " + latex);
+    }
+
+    return JSON.stringify(expression, serializedComponentsReplacer);
+}
+
+/**
+ * Return a LaTeX string that corresponds to a mathematical expression.
+ */
+export function toLatex(
+    mathObject: string,
+    padToDecimals?: number,
+    padToDigits?: number,
+    showBlanks?: boolean,
+) {
+    let mathExpr = JSON.parse(mathObject, serializedComponentsReviver);
+
+    return mathExpr.toLatex({
+        padToDecimals,
+        padToDigits,
+        showBlanks,
+    });
+}
+
+/**
+ * Create a new mathematical expression formed by substituting variables with new expressions
+ */
+export function substituteIntoMath(
+    mathObject: string,
+    substitutions: Map<string, any>,
+) {
+    let mathExpr = JSON.parse(mathObject, serializedComponentsReviver);
+
+    // math-expressions substitute expects an object, not a Map
+    let subs_object = {};
+    for (const [key, value] of substitutions.entries()) {
+        subs_object[key] = JSON.parse(value, serializedComponentsReviver);
+    }
+
+    let newExpr = mathExpr.substitute(subs_object);
+
+    return JSON.stringify(newExpr, serializedComponentsReplacer);
+}
+
+enum Simplify {
```

Comment:
Don't use typescript enums. Declare these as consts or as a const object (e.g., `const SIMPLIFY = {...} as const;`

Reply:
```typescript
const SIMPLIFY = {
    none: "none",
    numbersPreserveOrder: "numberspreserveorder",
    numbers: "numbers",
    full: "full"
} as const;
```

Reply:
Sure, but `simplify` is a `string` that can take on only one of those four values. (It came from a Rust enum.) Just making sure that assigning the const `SIMPLIFY` as the type for `simplify` has that meaning. It has a different feel to me, but I haven't taken the time to learn TypeScript in any depth.

Reply:
Oh, I see what you're saying. Just make the type in the simplify function `option: "none" | "numberspreserveorder" | "numbers" | "full"` That will prevent a typescript user from mistyping it :-P